spec/yuriita/query_spec.rb

```diff
@@ -16,4 +16,115 @@
       expect(query.inputs).to eq inputs
     end
   end
+
+  describe "#dup" do
+    it "duplicates the inputs" do
+      input = Yuriita::Query::Input.new(qualifier: "is", term: "original")
+      new_input = Yuriita::Query::Input.new(qualifier: "is", term: "new")
+      original = described_class.new(inputs: [input])
+
+      duplicate = original.dup
+      duplicate << new_input
+
+      expect(duplicate.inputs).to include(input)
+      expect(original.inputs).not_to include(new_input)
+    end
+  end
+
+  describe "#clone" do
+    it "duplicates the inputs" do
+      input = Yuriita::Query::Input.new(qualifier: "is", term: "original")
+      new_input = Yuriita::Query::Input.new(qualifier: "is", term: "new")
+      original = described_class.new(inputs: [input])
+
+      duplicate = original.clone
+      duplicate << new_input
+
+      expect(duplicate.inputs).to include(input)
+      expect(original.inputs).not_to include(new_input)
+    end
+  end
+
+  describe "#each" do
+    it "yields the inputs" do
+      published = build(:input, qualifier: "is", term: "published")
+      draft = build(:input, qualifier: "is", term: "draft")
+      query = described_class.new(inputs: [published, draft])
+
+      expect do |b|
+        query.each(&b)
+      end.to yield_successive_args(published, draft)
+    end
+  end
+
+  describe "each_element" do
+    it "yields inputs and keywords" do
```

Comment:
any reason this is "each_element" and not "#each_element"?

Reply:
Nope, good catch. Pushed a fix for `each_element`, along with `delete_if` and `size`.